chromadb/test/conftest.py

```diff
@@ -651,13 +681,73 @@ def api(system: System) -> Generator[ServerAPI, None, None]:
         yield api
 
 
+class ClientFactories:
+    """This allows consuming tests to be parameterized by async/sync versions of the client and papers over the async implementation.
+    If you don't need to manually construct clients, use the `client` fixture instead.
+    """
+
+    _system: System
+    _created_clients: List[ClientAPI] = []
+
+    def __init__(self, system: System):
+        self._system = system
+
+    def create_client(self, *args: Any, **kwargs: Any) -> ClientCreator:
+        if kwargs.get("settings") is None:
+            kwargs["settings"] = self._system.settings
+
+        if (
+            self._system.settings.chroma_api_impl
+            == "chromadb.api.async_fastapi.AsyncFastAPI"
+        ):
+            client = cast(ClientCreator, AsyncClientCreatorSync.create(*args, **kwargs))
+            self._created_clients.append(client)
+            return client
+
+        client = ClientCreator(*args, **kwargs)
+        self._created_clients.append(client)
```

Comment:
Why do we need to hang on to these and manually delete them - is it just to call clear_system_cache? Can we document that.